plugins/catalog-backend-module-gitlab/src/lib/types.ts

```diff
@@ -31,12 +31,41 @@ export type GitLabProject = {
   path_with_namespace?: string;
 };
 
+export type GitLabUser = {
+  id: number;
+  username: string;
+  name: string;
+  email: string;
```

Comment:
When I use my personal, [non-administrator](https://docs.gitlab.com/ee/api/users.html#for-non-administrator-users) access token to read users, my response only includes the fields `id`, `username`, `name`, `state`, `avatar_url`, and `web_url`.

Do we want this integration to require an admin token? IMHO I'd rather we didn't, since that's a level of privilege many regulated orgs wouldn't want to give to Backstage, but if we do we should call it out in the docs!

I'm particularly interested in this `email` field, which is used for sign-in by the `commonByEmailResolver`:

https://github.com/backstage/backstage/blob/3a373776e628bd93783ff3baaf2b0d6efe7c531c/plugins/auth-backend/src/providers/resolvers.ts#L39-L58

Reply:
Interesting. I am an Admin on our Instance, I didn't know it gives different responses for admins and non-admins. One way we can workaround the email issue:
 - Simply don't get the email and use the username as "login" identifier
 - Querying the detail for every user and check if he has a public email defined. Every user must configure that for himself though.
 - Use Admin Token.

I'll go ahead and implement the first approach for now and add some documentation to it.

Reply:
It needs a Admin PAT, otherwise the memberships API is not allowed.

Reply:
That explains why I was able to ingest users but not groups!

Reply:
Can we make sure that's clearly documented? At least in the doc comment to the relevant public class, but also in the appropriate docs section on backstage.io if there is one

Reply:
https://github.com/backstage/backstage/pull/15888/files#diff-ef9e9afa66ac0898509f0f03fedab0225b6766e605e3b3e53280519d4be195efR25